### PR TITLE
Updating the script to use the AzureAD PowerShell module instead of the AzureRM one

### PR DESCRIPTION
**Pull request by jmprieur**
With this change it's now possible to infer the tenant name (xxx.onmicrosoft.com) from the Tenant Id.
Also fixes a typo in a variable name

### CODE REVIEW

File: Configure.ps1 [hunk 1]
```diff
@@ -82,62 +81,63 @@ of the visual studio solution (App.Config and Web.Config) so that they are consi
...
+    # Add a Azure Key Credentials from the certificate for the application
+    $applicationKeyCredentials = New-AzureADApplicationKeyCredential -ObjectId $daemonApplication.ObjectId -CustomKeyIdentifier $certificateName -Type AsymmetricX509Cert -Usage Verify -Value $certBase64Value  -StartDate $certificate.NotBefore -EndDate $certificate.NotAfter
 
     # Update the config file in the daemon application
     $configFile = $pwd.Path + "\TodoListDaemonWithCert\App.Config"
     Write-Host "Updating the configuration file for the client ($configFile)"
-    UpdateClientConfigFile -configFilePath $configFile -tenantId $tenantId -clientId $daemonApplication.ApplicationId -appUri $serviceAppIdIRI -baseAddress $serviceHomePage -certificateName $certificateName
+    UpdateClientConfigFile -configFilePath $configFile -tenantId $tenantId -clientId $daemonApplication.AppId -appUri $serviceAppIdURI -baseAddress $serviceHomePage -certificateName $certificateName
 
     # Update tehe config file in the service application
```
Comment (blairmcg) [inline Configure.ps1]: typo

File: Configure.ps1 [hunk 2]
```diff
@@ -146,10 +146,10 @@ of the visual studio solution (App.Config and Web.Config) so that they are consi
 #Clean-Ups the applications and serice principal in Azure AD.
```
Comment (blairmcg) [inline Configure.ps1]: Two typos ("clean-ups" and "serice"): "Remove the applications and service principals registered for the sample", or "Remove the application registrations previously created for the sample"

File: Configure.ps1 [hunk 3]
```diff
@@ -4,8 +4,8 @@
 # Before running this script you need to install the Azure RM cmdlets as an administrator. 
 # For this:
 # 1) Run Powershell as an administrator
-# 2) in the PowerShell window, type: Install-Module AzureRM.Resources
-#
+# 2) in the PowerShell window, type: Install-Module AzureAD
+
 # To prepare this script
 # 3) With the Azure portal (https://portal.azure.com), choose your active directory tenant, then go to the Properties of the tenant and copy
 #    the DirectoryID. This is what we'll use in this script for the tenant ID
```
Comment (blairmcg) [inline Configure.ps1]: Are you going to update the readme to explain how to use this script, including using the cleanup function when done?

File: Configure.ps1 (hunk 1 above)
Comment (jmprieur) [inline Configure.ps1]: where? on line 132 ?

File: Configure.ps1 (hunk 3 above)
Comment (jmprieur) [inline Configure.ps1]: Yes, Blair. My intention was to:
- change the Readme.md to extract the manual process to another .md file (for people who want to understand how to do that manually)
- replace the content of the manual process by explanations on how to run the script, and how to cleanup.

Comment (bgavrilMS) [inline Configure.ps1]: In Powershell, do not call functions using paranthesis - i.e. cleanup($apps)  on line 22. Instead, call it with cleanup $apps. Also, to be consistent with the other functions - please use Cleanup.

File: Configure.ps1 [hunk 4]
```diff
@@ -4,8 +4,8 @@
 # Before running this script you need to install the Azure RM cmdlets as an administrator.
```
Comment (mjcheetham) [inline Configure.ps1]: You should add a line to import the correct PowerShell module. That way the script will fail upfront if the module is not found.

File: Configure.ps1 [hunk 5]
```diff
@@ -82,74 +81,75 @@ of the visual studio solution (App.Config and Web.Config) so that they are consi
...
+         $tenantId = $creds.Tenant.Id
+     }
+    $tenant = Get-AzureADTenantDetail
+    $tenantName =  $tenant.VerifiedDomains[0].Name
 
     # Variables for the registration of the AAD application for the Web API Service
     $serviceAadAppName = "TodoListService"
     $serviceHomePage = "https://localhost:44321"
-    $serviceAppIdIRI = "https://"+$tenantName+"/"+$serviceAadAppName
+    $serviceAppIdURI = "https://"+$tenantName+"/"+$serviceAadAppName
```
Comment (mjcheetham) [inline Configure.ps1]: In PowerShell, strings in double-quotes `"` are interpolated.

```powershell
$serviceAppIdURI = "https://$tenantName/$serviceAadAppName"
```

If you want to use a block of code, or refer to a variable which uses a colon `:` then you can encompass it with the `$` character. For example:

```powershell
$script:myScriptVar = 3
$otherVar = "My favourite number is: ${script:myScriptVar}."
```

File: Configure.ps1 [hunk 6]
```diff
@@ -82,74 +81,75 @@ of the visual studio solution (App.Config and Web.Config) so that they are consi
...
-#Clean-Ups the applications and serice principal in Azure AD.
-Function CleanUp($applicationsInformation)
+# Remove the AAD applications and service principals registered for the sample.
+Function Cleanup($applicationsInformation)
 {
-    remove-azurermadserviceprincipal -objectid $applicationsInformation.serviceAppServicePrincipal.id -force
-    remove-azurermadapplication -objectid $applicationsInformation.serviceApp.objectid -force
-    remove-azurermadserviceprincipal -objectid $applicationsInformation.clientAppServicePrincipal.id -force
-    remove-azurermadapplication -objectid $applicationsInformation.clientApp.objectid -force
+    remove-azurermadserviceprincipal -objectid $applicationsInformation.serviceAppServicePrincipal.id
```
Comment (mjcheetham) [inline Configure.ps1]: You should capitalise cmdlets names when invoking them as a general best practice.

`Remove-AzureADServicePrincipal`

Comment (mjcheetham) [inline Configure.ps1]: Same goes for variable names too. The PowerShell ISE or VS Code should autocomplete to the correct case them when pressing [Tab].

File: Configure.ps1 [hunk 7]
```diff
@@ -82,74 +81,75 @@ of the visual studio solution (App.Config and Web.Config) so that they are consi
     # Active Directory Tenant. This is a GUID which represents the "Directory ID" of the AzureAD tenant into which you want to create the apps.
     # Look it up in the Azure portal in the "Properties" of the Azure AD. 
     # IN THE FUTURE: We'd like to be able to get the tenant name from the tenant ID (or pass the tenant name as an argument)
-    $tenantName = $tenantId
+
+    # Login to Azure PowerShell (interactive: you'll need to sign-in with creds enabling your to create apps in the tenant)
+    $creds = Connect-AzureAD -TenantId $tenantId
+     if (!$tenantId)
```
Comment (mjcheetham) [inline Configure.ps1]: Is there a reason why this `if` block is indented by 1 character from the surrounding lines?

File: Configure.ps1 [hunk 8]
```diff
@@ -82,74 +81,75 @@ of the visual studio solution (App.Config and Web.Config) so that they are consi
...
-    $keyCredential.EndDate= $certificate.NotAfter
-
-    # And create the daemon application
-    $daemonApplication = New-AzureRmADApplication -DisplayName $daemonAadAppName -HomePage $daemonHomePage -IdentifierUris $daemonAppIdIRI -KeyCredentials $keyCredential
-    $daemonservicePrincipal = New-AzureRmADServicePrincipal -ApplicationId $daemonApplication.ApplicationId
+    $certBase64Value = [System.Convert]::ToBase64String($certificate.GetRawCertData())
+    $certBase64Thumbprint = [System.Convert]::ToBase64String($certificate.GetCertHash())
+    
+    # Add a Azure Key Credentials from the certificate for the application
+    $applicationKeyCredentials = New-AzureADApplicationKeyCredential -ObjectId $daemonApplication.ObjectId -CustomKeyIdentifier $certificateName -Type AsymmetricX509Cert -Usage Verify -Value $certBase64Value  -StartDate $certificate.NotBefore -EndDate $certificate.NotAfter
```
Comment (mjcheetham) [inline Configure.ps1]: If you want to make this a bit more readable then you can split the lines using the backtick ` character.

```
$applicationKeyCredentials = New-AzureADApplicationKeyCredential -ObjectId $daemonApplication.ObjectId `
                                                                 -CustomKeyIdentifier $certificateName `
                                                                 -Type AsymmetricX509Cert `
                                                                 -Usage Verify `
                                                                 -Value $certBase64Value  `
                                                                 -StartDate $certificate.NotBefore `
                                                                 -EndDate $certificate.NotAfter
```

File: Configure.ps1 (hunk 4 above)
Comment (jmprieur) [inline Configure.ps1]: @mjcheetham : Done

File: Configure.ps1 (hunk 5 above)
Comment (jmprieur) [inline Configure.ps1]: Thanks @mjcheetham : Done

File: Configure.ps1 (hunk 6 above)
Comment (jmprieur) [inline Configure.ps1]: Fixed

File: Configure.ps1 (hunk 7 above)
Comment (jmprieur) [inline Configure.ps1]: no, that was a mistake

File: Configure.ps1 (hunk 8 above)
Comment (jmprieur) [inline Configure.ps1]: Thanks for the tip. Fixed. This is much more readable now.

**Pull request merged**